File: CS/PageViewSample/DataStorage.cs

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Media.Imaging;
using System.Xml.Serialization;

namespace PageViewSample {
    [XmlRoot("Employees")]
    public class Employees : List<Employee> {
    }
    [XmlRoot("Employee")]
    public class Employee {
        public int Id { get; set; }
        public int ParentId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string FullName {
            get {
                return FirstName + " " + LastName;
            }
        }
        public string JobTitle { get; set; }
        public string Phone { get; set; }
        public string EmailAddress { get; set; }
        public string AddressLine1 { get; set; }
        public string City { get; set; }
        public string PostalCode { get; set; }
        public string CountryRegionName { get; set; }
        public string GroupName { get; set; }
        public DateTime BirthDate { get; set; }
        public DateTime HireDate { get; set; }
        public string Gender { get; set; }
        public string MaritalStatus { get; set; }
        public string Title { get; set; }
        public byte[] ImageData { get; set; }
        BitmapImage imageSource;
        [XmlIgnore]
        public BitmapImage Photo {
            get {
                if (imageSource == null)
                    SetImageSource();
                return imageSource;
            }
        }
        void SetImageSource() {
            var stream = new MemoryStream(ImageData);
            imageSource = new BitmapImage();
            imageSource.BeginInit();
            imageSource.StreamSource = stream;
            imageSource.EndInit();
        }
    }

    public static class DataStorage {
        static Employees employees;
        public static Employees Employees {
            get {
                if (employees == null) {
                    var stream = System.Reflection.Assembly.GetEntryAssembly().GetManifestResourceStream("PageViewSample.Employees.xml");
                    XmlSerializer serializier = new XmlSerializer(typeof(Employees));
                    employees = serializier.Deserialize(stream) as Employees;
                }
                return employees;
            }
        }
    }
    public class EmployeesData {
        public Employees DataSource {
            get { return DataStorage.Employees; }
        }
    }
}

File: VB/PageViewSample/DataStorage.vb
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Windows.Media.Imaging
Imports System.Xml.Serialization

Namespace PageViewSample

    <XmlRoot("Employees")>
    Public Class Employees
        Inherits List(Of Employee)

    End Class

    <XmlRoot("Employee")>
    Public Class Employee

        Public Property Id As Integer

        Public Property ParentId As Integer

        Public Property FirstName As String

        Public Property LastName As String

        Public ReadOnly Property FullName As String
            Get
                Return FirstName & " " & LastName
            End Get
        End Property

        Public Property JobTitle As String

        Public Property Phone As String

        Public Property EmailAddress As String

        Public Property AddressLine1 As String

        Public Property City As String

        Public Property PostalCode As String

        Public Property CountryRegionName As String

        Public Property GroupName As String

        Public Property BirthDate As Date

        Public Property HireDate As Date

        Public Property Gender As String

        Public Property MaritalStatus As String

        Public Property Title As String

        Public Property ImageData As Byte()

        Private imageSource As BitmapImage

        <XmlIgnore>
        Public ReadOnly Property Photo As BitmapImage
            Get
                If imageSource Is Nothing Then SetImageSource()
                Return imageSource
            End Get
        End Property

        Private Sub SetImageSource()
            Dim stream = New MemoryStream(ImageData)
            imageSource = New BitmapImage()
            imageSource.BeginInit()
            imageSource.StreamSource = stream
            imageSource.EndInit()
        End Sub
    End Class

    Public Module DataStorage

        Private employeesField As Employees

        Public ReadOnly Property Employees As Employees
            Get
                If employeesField Is Nothing Then
                    Dim stream = Reflection.Assembly.GetEntryAssembly().GetManifestResourceStream("PageViewSample.Employees.xml")
                    Dim serializier As XmlSerializer = New XmlSerializer(GetType(Employees))
                    employeesField = TryCast(serializier.Deserialize(stream), Employees)
                End If

                Return employeesField
            End Get
        End Property
    End Module

    Public Class EmployeesData

        Public ReadOnly Property DataSource As Employees
            Get
                Return DataStorage.Employees
            End Get
        End Property
    End Class
End Namespace
